source/distributed/tria.cc

```diff
@@ -3594,16 +3594,13 @@ namespace parallel
           types::subdomain_id                                  ghost_owner = 0;
           typename dealii::internal::p4est::types<dim>::topidx ghost_tree  = 0;
 
-          for (unsigned int g_idx = 0;
-               g_idx < parallel_ghost->ghosts.elem_count;
+          for (int g_idx = 0; static_cast<unsigned int>(g_idx) <
+                              parallel_ghost->ghosts.elem_count;
```

Comment:
why cast it this way around? `g_idx` is logically an unsigned quantity. The fact that p4est stores certain data as signed ints does not change this.

Reply:
Less casts improve the readability, give less lines of code and lead to nicer indenting.
It seems I underestimated the significance of unsigned types vs. signed types here. Let me just change it back.

In the end, there is (apart from the one bit more space) really not much of an advantage of using unsigned types if we don't care for defined behavior for overflow... Of course, we should just be consistent and that is the style we have chosen.

Reply:
Yes, we're really consistent everywhere in using unsigned data types for things that are logically unsigned. That's not about the overflow behavior but just attaching additional semantic meanings to variables: if you have an index or a count, it should be unsigned.